fem/bilinearform.cpp

```diff
@@ -1742,6 +1862,101 @@ void MixedBilinearForm::ComputeBdrElementMatrix(int i, DenseMatrix &elmat)
    }
 }
 
+void MixedBilinearForm::ComputeFaceMatrix(int i, DenseMatrix &elmat) const
+{
+   FaceElementTransformations *ftr;
+   Mesh *mesh = test_fes -> GetMesh();
+   ftr = mesh->GetFaceElementTransformations(i);
+   MFEM_ASSERT(ftr, "No associated face transformations.");
+
+   if (interior_face_integs.Size())
+   {
+      const FiniteElement *trial_fe1, *trial_fe2, *test_fe1, *test_fe2;
+
+      trial_fe1 = trial_fes->GetFE(ftr->Elem1No);
+      test_fe1 = test_fes->GetFE(ftr->Elem1No);
+      if (ftr->Elem2No >= 0)
+      {
+         trial_fe2 = trial_fes->GetFE(ftr->Elem2No);
+         test_fe2 = test_fes->GetFE(ftr->Elem2No);
+      }
+      else
+      {
+         // The test_fe2 object is really a dummy and not used on the
+         // boundaries, but we can't dereference a NULL pointer, and we don't
+         // want to actually make a fake element.
+         trial_fe2 = trial_fe1;
+         test_fe2 = test_fe1;
+      }
+
```

Comment:
You could precompute the size of elmat so you don't have to fill the vdof arrays in the else case, i.e you could use something like 
````cpp
   ntrial = trial_fe1->GetDof() * trial_fes->GetVdim();
   ntest = test_fe1->GetDof() * test_fes->GetVdim();
   if (ftr->Elem2No >= 0)
   {
      ntrial += trial_fe2->GetDof() * trial_fes->GetVdim();
      ntest += test_fe2->GetDof() * test_fes->GetVdim();
   }
````

Reply:
Just a logic copied from `MixedBilinearForm::ComputeElementMatrix()`, but it definitely makes sense and we may change it everywhere 😉 .

Reply:
Whoops, it seems I added those methods some years ago to MFEM 😅 But I probably copied that from somewhere else even back then as I know myself... 😄

Reply:
Yes, version 1.0 as above 👍 . I will change it in `BilinearForm` too.

Reply:
Ok, I have left a bit of super optimization by not prefilling the matrix when integrators exist, but the empty matrices are made without that VDOFs arrays as you wanted 👍 .